packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/IncrementalLayoutEngine.java

```diff
@@ -78,8 +79,11 @@ public class IncrementalLayoutEngine {
 
     private final NodeSizeProvider nodeSizeProvider;
 
-    public IncrementalLayoutEngine(NodeSizeProvider nodeSizeProvider) {
+    private final List<ICustomNodeLabelPositionProvider> customLabelPositionProviders;
+
+    public IncrementalLayoutEngine(NodeSizeProvider nodeSizeProvider, List<ICustomNodeLabelPositionProvider> customLabelPositionProviders) {
         this.nodeSizeProvider = Objects.requireNonNull(nodeSizeProvider);
+        this.customLabelPositionProviders = customLabelPositionProviders;
```

Comment:
what if no contributions? The list is null or empty?

Reply:
The requireNonNull is here to prevent the list to be `null`, but it can be empty.
If the list is `null`, you must check if the list is not null before iterating it. `l.169` there is no check for that.

Reply:
I added requireNonNull